app/src/components/integrations/integrationProviders/ldapIntegration/ldapFormFields/ldapFormFields.jsx

```diff
@@ -0,0 +1,334 @@
+import { injectIntl, defineMessages, intlShape } from 'react-intl';
+import { formValueSelector } from 'redux-form';
+import { connect } from 'react-redux';
+import classNames from 'classnames/bind';
+import { FieldErrorHint } from 'components/fields/fieldErrorHint';
+import { Input } from 'components/inputs/input';
+import { InputDropdown } from 'components/inputs/inputDropdown';
+import { InputConditional } from 'components/inputs/inputConditional';
+import { IntegrationFormField, INTEGRATION_FORM } from 'components/integrations/elements';
+import { validate, commonValidators, bindMessageToValidator, isEmptyObject } from 'common/utils';
+import {
+  LDAP_ATTRIBUTES_KEY,
+  LDAP_PREFIX,
+  LDAPS_PREFIX,
+  URL_KEY,
+  BASE_DN_KEY,
+  PHOTO_KEY,
+  FULL_NAME_KEY,
+  EMAIL_KEY,
+} from 'pages/admin/serverSettingsPage/common/constants';
+import {
+  DEFAULT_FORM_CONFIG,
+  MANAGER_DN_KEY,
+  GROUP_SEARCH_BASE_KEY,
+  GROUP_SEARCH_FILTER_KEY,
+  USER_SEARCH_FILTER_KEY,
+  USER_DN_PATTERN_KEY,
+  PASSWORD_ATTRIBUTE_KEY,
+  PASSWORD_ENCODER_TYPE_KEY,
+  MANAGER_PASSWORD_KEY,
+} from '../constants';
+import styles from './ldapFormFields.scss';
+
+const cx = classNames.bind(styles);
+
+const messages = defineMessages({
+  urlLabel: {
+    id: 'LdapFormFields.urlLabel',
+    defaultMessage: 'Url',
+  },
+  baseDnLabel: {
+    id: 'LdapFormFields.baseDnLabel',
+    defaultMessage: 'Base DN',
+  },
+  managerDnLabel: {
+    id: 'LdapFormFields.managerDnLabel',
+    defaultMessage: 'Manager DN',
+  },
+  managerPasswordLabel: {
+    id: 'LdapFormFields.managerPasswordLabel',
+    defaultMessage: 'Manager password',
+  },
+  userDnPatternLabel: {
+    id: 'LdapFormFields.userDnPatternLabel',
+    defaultMessage: 'User DN pattern',
+  },
+  userSearchFilterLabel: {
+    id: 'LdapFormFields.userSearchFilterLabel',
+    defaultMessage: 'User search filter',
+  },
+  groupSearchBaseLabel: {
+    id: 'LdapFormFields.groupSearchBaseLabel',
+    defaultMessage: 'Group search base',
+  },
+  groupSearchFilterLabel: {
+    id: 'LdapFormFields.groupSearchFilterLabel',
+    defaultMessage: 'Group search filter',
+  },
+  passwordEncoderTypeLabel: {
+    id: 'LdapFormFields.passwordEncoderTypeLabel',
+    defaultMessage: 'Password encoder type',
+  },
+  passwordAttributeLabel: {
+    id: 'LdapFormFields.passwordAttributeLabel',
+    defaultMessage: 'Password attribute',
+  },
+  emailAttributeLabel: {
+    id: 'LdapFormFields.emailAttributeLabel',
+    defaultMessage: 'Email attribute',
+  },
+  fullNameAttributeLabel: {
+    id: 'LdapFormFields.fullNameAttributeLabel',
+    defaultMessage: 'Full name attribute',
+  },
+  photoAttributeLabel: {
+    id: 'LdapFormFields.photoAttributeLabel',
+    defaultMessage: 'Photo attribute',
+  },
+});
+
+const urlValidator = bindMessageToValidator(validate.ldapUrl, 'requiredFieldHint');
+
+@connect((state) => ({
+  passwordEncoderType: formValueSelector(INTEGRATION_FORM)(state, PASSWORD_ENCODER_TYPE_KEY),
```

Comment:
You can move `formValueSelector(INTEGRATION_FORM)` out of this function

Reply:
Fixed